File: frontend/src/types/revenue.ts
export interface RevenueRecord {
  id: string
  user_id: string | null
  income_category: string
  amount: number
  currency: string
  platform: string | null
  source_description: string | null
  recorded_date: string | null
  is_verified: boolean
  created_at: string | null
}

export interface IncomeCategoryInfo {
  name: string
  color: string
  amount: number
  proportion: number
}

export interface DiversityIndexResponse {
  diversity_index: number
  total_sources: number
  warnings: string[]
  category_distribution: Record<string, IncomeCategoryInfo>
}

export interface RevenueSummaryResponse {
  user_id: string
  total_revenue: number
  currency: string
  months: number
  monthly_trend: Array<{ month: string; amount: number }>
  diversity: DiversityIndexResponse
}

export type IncomeCategoryKey =
  | 'ad_revenue'
  | 'sponsorship'
  | 'subscription'
  | 'tip'
  | 'ecommerce'
  | 'affiliate'
  | 'knowledge_payment'
  | 'ip_licensing'

File: backend/app/schemas/revenue.py
"""收入多元化分析 Pydantic schemas."""

from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class RevenueRecordCreate(BaseModel):
    income_category: str
    amount: float
    currency: str = "CNY"
    platform: Optional[str] = None
    source_description: Optional[str] = None
    recorded_date: Optional[datetime] = None


class RevenueRecordSchema(BaseModel):
    id: str
    user_id: Optional[str] = None
    income_category: str
    amount: float
    currency: str
    platform: Optional[str]
    source_description: Optional[str]
    recorded_date: Optional[datetime]
    is_verified: bool
    created_at: Optional[datetime] = None


class DiversityIndexResponse(BaseModel):
    diversity_index: float
    total_sources: int
    warnings: list[str]
    category_distribution: dict


class RevenueSummaryResponse(BaseModel):
    user_id: Optional[str] = None
    total_revenue: float
    currency: str
    months: int
    monthly_trend: list[dict]
    diversity: DiversityIndexResponse
